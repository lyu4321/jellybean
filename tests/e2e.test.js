const run = require('./run');

describe('E2E integration', () => {
    test('Print error message when no input', async () => {
        const { stderr, stdout, exitCode } = await run();
        expect(exitCode).toBe(1);
        expect(stderr).toMatchSnapshot();
        expect(stdout).toEqual('');
    });

    test('Prints help message when --help flag used', async () => {
        const { stderr, stdout, exitCode } = await run('--help');
        expect(exitCode).toBe(0);
        expect(stdout).toMatchSnapshot();
        expect(stderr).toEqual('');
    });

    test('Prints version when --version flag used', async () => {
        const { stderr, stdout, exitCode } = await run('--version');
        expect(exitCode).toBe(0);
        expect(stdout).toMatchSnapshot();
        expect(stderr).toEqual('');
    });

    test('Prints error message when invalid input', async () => {
        const { stderr, stdout, exitCode } = await run(
            '--input',
            'invalidinput'
        );
        expect(exitCode).toBe(1);
        expect(stderr).toMatchSnapshot();
        expect(stdout).toEqual('');
    });

    test('Prints error message when empty folder', async () => {
        const { stderr, stdout, exitCode } = await run(
            '--input',
            'invalid-folder'
        );
        expect(exitCode).toBe(1);
        expect(stderr).toMatchSnapshot();
        expect(stdout).toEqual('');
    });

    test('Generates HTML file when valid file path', async () => {
        const { stderr, stdout, exitCode } = await run(
            '--input',
            'test-docs/Silver Blaze.txt'
        );
        expect(exitCode).toBe(0);
        expect(stdout).toMatchSnapshot();
        expect(stderr).toEqual('');
    });

    test('Generates HTML files when valid folder path', async () => {
        const { stderr, stdout, exitCode } = await run('--input', 'test-docs');
        expect(exitCode).toBe(0);
        expect(stdout).toMatchSnapshot();
        expect(stderr).toEqual('');
    });

    test('Generates HTML files when valid file and output paths', async () => {
        const { stderr, stdout, exitCode } = await run(
            '--input',
            'test-docs',
            '--output',
            'empty-folder'
        );
        expect(exitCode).toBe(0);
        expect(stdout).toMatchSnapshot();
        expect(stderr).toEqual('');
    });
});
