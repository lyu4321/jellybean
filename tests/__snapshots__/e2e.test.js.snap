// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E2E integration Generates HTML file when valid file path 1`] = `"The file dist/Silver Blaze.html has been created"`;

exports[`E2E integration Generates HTML files when valid file and output paths 1`] = `
"The file empty-folder/Silver Blaze.html has been created
The file empty-folder/The Adventure of the Six Napoleans.html has been created"
`;

exports[`E2E integration Generates HTML files when valid folder path 1`] = `
"The file dist/Silver Blaze.html has been created
The file dist/The Adventure of the Six Napoleans.html has been created"
`;

exports[`E2E integration Print error message when no input 1`] = `
"Input file or folder is required
Please see --help for options."
`;

exports[`E2E integration Prints error message when empty folder 1`] = `"Input directory is empty."`;

exports[`E2E integration Prints error message when invalid input 1`] = `
"Input file or folder is required
Please see --help for options."
`;

exports[`E2E integration Prints help message when --help flag used 1`] = `
"Options:
  -h, --help        Show help                                          [boolean]
  -c, --config      Path to a JSON config file                          [string]
  -i, --input       Path to a file or folder with files                  [array]
  -l, --lang        Language used in generated HTML files               [string]
  -o, --output      Path to an output directory                         [string]
  -s, --stylesheet  Stylesheet URL                                      [string]
  -v, --version     Show version number                                [boolean]"
`;

exports[`E2E integration Prints version when --version flag used 1`] = `"jellybean 1.0.0"`;
